short_answer/static/js/src/short_answer.js

```diff
@@ -185,8 +206,26 @@ function ShortAnswerXBlock(runtime, element, options) {
     });
   });
 
+  /**
+   * Enable/Disable the submit button.
+   */
+  $shortAnswer.keyup(function() {
+    if (isSubmitAllowed()) {
+      $answerSubmitButton.removeAttr('disabled');
+    }
+    else {
+      $answerSubmitButton.attr('disabled', true);
+    }
```

Comment:
$answerSubmitButton.attr('disabled', !isSubmitAllowed());

Perhaps changing the implementation to actually check if submit is disallowed would decrease the need for two ! operators and would increase readability slightly.